src/server/v2.0/handler/artifact.go

```diff
@@ -103,11 +85,98 @@ func (a *artifactAPI) ListArtifacts(ctx context.Context, params operation.ListAr
 }
 
 func (a *artifactAPI) GetArtifact(ctx context.Context, params operation.GetArtifactParams) middleware.Responder {
-	// TODO implement
-	return operation.NewGetArtifactOK()
+	// set option
+	option := option(params.WithTag, params.WithImmutableStatus,
+		params.WithLabel, params.WithScanOverview, params.WithSignature)
+
+	// get the artifact
+	artifact, err := a.artCtl.GetByReference(ctx, fmt.Sprintf("%s/%s", params.ProjectName, params.RepositoryName), params.Reference, option)
+	if err != nil {
+		return a.SendError(ctx, err)
+	}
+	return operation.NewGetArtifactOK().WithPayload(artifact.ToSwagger())
 }
 
 func (a *artifactAPI) DeleteArtifact(ctx context.Context, params operation.DeleteArtifactParams) middleware.Responder {
-	// TODO implement
+	artifact, err := a.artCtl.GetByReference(ctx, fmt.Sprintf("%s/%s", params.ProjectName, params.RepositoryName), params.Reference, nil)
+	if err != nil {
+		return a.SendError(ctx, err)
+	}
```

Comment:
what about to remove the query and call the delete directly?

Reply:
Deletion is only supported with ID for now.

Reply:
Why not let the caller provide the ID? That means the caller can call some `get`/`list` method to have the related ID and then trigger deletion with that ID.

Reply:
This is in the API handler already, there is no upper caller